### PR TITLE
Update html guidelines

### PR DESCRIPTION
**Pull request by wbamberg**
This PR updates the contribution docs for HTML examples, `CONTRIBUTING-HTML.md` to add more detailed and (hopefully) useful guidelines.

It restructures that file into 3 main sections:

1. a walkthrough of how to add an example
2. a set of guidelines for writing good examples
3. a bunch of extra bits and pieces (changing editor height, encoding entities, fonts)

The set of guidelines is taken from the gist at https://gist.github.com/wbamberg/355a58406c01dc9ec3629fe0c69ff580, with some small updates based on review comments.

### CODE REVIEW

File: CONTRIBUTING-HTML.md [hunk 1]
```diff
@@ -219,3 +146,198 @@ Once you've finished writing the example, see the [Testing](CONTRIBUTING.md#test
...
+This section describes some guidelines to follow when writing HTML examples. It's split into two sections:
+
+* **Formal guidelines** cover formal aspects of the example, such as how long it should be.
+* **Content guidelines** cover the actual content of the example: that is, what it should include or exclude.
+
+Sometimes formal and content guidelines are at odds. For example, sometimes writing a useful example means making it longer than the formal guidelines ask. Usually, when this happens, we should prioritize content guidelines.
+
+### Formal guidelines
+
+In general: try out your example using `npm run start` and see what it looks like with a browser window width of 1000px.
```
Comment (schalkneethling) [inline CONTRIBUTING-HTML.md]: Nit: You can also just run `npm start`

File: CONTRIBUTING-HTML.md [hunk 2]
```diff
@@ -219,3 +146,198 @@ Once you've finished writing the example, see the [Testing](CONTRIBUTING.md#test
...
+</p>
+
+<p><button accesskey="a">Button with <u>A</u>ccesskey</button></p>
+
+<p><button class="styled">Fancy styled button</button></p>
+```
+
+This includes 5 different button elements. It's too long. The example should include a single element with some common attributes.
+
+Try out your example with `npm run start`. Can you immediately see the point of the example, or is the element buried in a wall of source?
```
Comment (schalkneethling) [inline CONTRIBUTING-HTML.md]: Nit: You can also just run `npm start`

File: CONTRIBUTING-HTML.md (hunk 1 above)
Comment (wbamberg) [reply CONTRIBUTING-HTML.md]: all these months I've been typing those extra 4 characters! :)

**Comment by wbamberg** [comment]
@chrisdavidmills , I know you saw the standalone version of this but I wanted you to have the chance to comment on this PR too. But if you don't think it's necessary please remove your review request and I'll merge it.

**Comment by atopal** [comment]
Thanks for putting this together, Will!

**Pull request merged**